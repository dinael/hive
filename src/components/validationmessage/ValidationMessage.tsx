import { FC } from 'react'
import styles from './ValidationMessage.module.scss'
import Icon from '../icon/Icon'

export type ValidationMessageProps = {
  message: string
  kind:  'info' | 'warning' | 'error' | 'success' | 'help' | 'safe'
  icon?: string
  id?: string
  role?: string
  className?: string
}

export const ValidationMessage: FC<ValidationMessageProps> = ({
  message,
  kind,
  id,
  icon,
  role,
  className,
  ...props
}: ValidationMessageProps) => {

  return (
    <span
      id={id ? id : ''}
      className={`
        ${styles['validation-message']}
        ${styles[kind]}
        ${className ? className : ''}}
      `}
      role={role ? role : ''}
      {...props}>
      {icon ? <Icon name={icon} size={'m'} /> : null}
      {message}
    </span>
  )
}

export default ValidationMessage
