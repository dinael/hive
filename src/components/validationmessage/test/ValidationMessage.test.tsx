import { describe, test, expect, vi } from 'vitest'
import { render } from '@testing-library/react'
import ValidationMessage, { ValidationMessageProps } from '../ValidationMessage'

describe('ValidationMessage', () => {
  const defaultProps: ValidationMessageProps = {
    message: 'message-test',
    kind: 'error',
    id: 'id-test',
    icon: 'close',
    role: 'alert',
    className: 'className-Text',
  }

  test('renders snapshot', () => {
    const { container } = render(<ValidationMessage {...defaultProps} />)
    expect(container).toMatchSnapshot()
  })
})
