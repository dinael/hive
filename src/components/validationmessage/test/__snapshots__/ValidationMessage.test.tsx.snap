// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ValidationMessage > ValidationMessage renders snapshot 1`] = `
<div>
  <span
    class="
    validation-message
    error
    className-Text}
  "
    id="id-test"
    role="alert"
  >
    <span
      aria-hidden="true"
      class="
    icon-component
    m
    
  "
      style="--icon-color: currentColor;"
    >
      <svg
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <title>
          close
        </title>
        <use
          xlink:href="/src/assets/sprite.svg#close"
        />
      </svg>
    </span>
    message-test
  </span>
</div>
`;
